src/Cli/func/Actions/LocalActions/PackAction/CustomPackSubcommandAction.cs

```diff
@@ -19,6 +22,80 @@ public async Task RunAsync(PackOptions packOptions)
             await ExecuteAsync(packOptions);
         }
 
+        protected override void ValidateFunctionApp(string functionAppRoot, PackOptions options)
+        {
+            PackValidationHelper.DisplayValidationStart();
+
+            // Validate that host.json exists
+            var hostJsonPath = Path.Combine(functionAppRoot, "host.json");
+            var hostJsonExists = FileSystemHelpers.FileExists(hostJsonPath);
+
+            PackValidationHelper.DisplayValidationResult(
+                "Validate Basic Structure",
+                hostJsonExists,
+                hostJsonExists ? null : "Required file 'host.json' not found. Ensure this is a valid Azure Functions project.");
+
+            if (!hostJsonExists)
+            {
+                PackValidationHelper.DisplayValidationEnd();
+                throw new CliException($"Required file 'host.json' not found in {functionAppRoot}. Custom handler function apps require host.json.");
+            }
+
+            // Validate custom handler configuration and executable
+            try
+            {
+                var hostJsonContent = FileSystemHelpers.ReadAllTextFromFileAsync(hostJsonPath).Result;
+                var hostConfig = JObject.Parse(hostJsonContent);
+
+                var customHandler = hostConfig["customHandler"];
+                if (customHandler != null)
+                {
+                    var description = customHandler["description"];
+                    if (description != null)
+                    {
+                        var defaultExecutablePath = description["defaultExecutablePath"]?.ToString();
+                        if (!string.IsNullOrEmpty(defaultExecutablePath))
+                        {
+                            var executablePath = Path.Combine(functionAppRoot, defaultExecutablePath);
+                            var executableExists = FileSystemHelpers.FileExists(executablePath);
+
+                            if (executableExists)
+                            {
+                                PackValidationHelper.DisplayValidationResult("Validate Custom Handler Executable", true);
+                            }
+                            else
+                            {
+                                PackValidationHelper.DisplayValidationWarning(
+                                    "Validate Custom Handler Executable",
+                                    $"Custom handler executable '{defaultExecutablePath}' not found. Ensure the executable exists or will be provided by other deployment methods.");
+                            }
+                        }
+                        else
+                        {
+                            PackValidationHelper.DisplayValidationResult("Validate Custom Handler Configuration", true, "No defaultExecutablePath specified in host.json");
```

Comment:
This test failed for me too. If i had empty string for "defaultExecutablePath". Func pack just ran fine without any issues.

<img width="1230" height="260" alt="image" src="https://github.com/user-attachments/assets/fc2bd5f7-12c1-4532-9304-c8551d18fa38" />